presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java

```diff
@@ -44,24 +44,108 @@
 public final class SystemPartitioningHandle
         implements ConnectorPartitioningHandle
 {
+    public static Partitioning singlePartition()
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.SINGLE);
+    }
+
+    public static boolean isSystemSinglePartitioning(Partitioning partitioning)
+    {
+        return isSystemPartitioning(partitioning, SystemPartitionFunction.SINGLE);
+    }
+
+    public static boolean isSystemSinglePartitioning(PartitioningHandle partitioningHandle)
+    {
+        return isSystemPartitioning(partitioningHandle.getConnectorHandle(), SystemPartitionFunction.SINGLE);
+    }
+
+    public static Partitioning coordinatorOnlyPartition()
+    {
+        return createSystemPartitioning(SystemPartitioning.COORDINATOR_ONLY, SystemPartitionFunction.SINGLE);
```

Comment:
The definition of a single partition implies partitioning on the empty of columns. We need to make sure that property is retained.